lib/filters/validator.js

```diff
@@ -48,9 +48,24 @@ var Validator = function(parameters) {
     this._paramCoercionFunc = this._createTypeCoercionFunc(parameters.filter(function(p) {
         return p.in !== 'formData' && p.in !== 'body' && p.type !== 'string';
     }));
-    this._bodyCoercionFunc = this._createTypeCoercionFunc(parameters.filter(function(p) {
-        return p.in === 'formData' && p.type !== 'string';
-    }));
+    var bodyParams = parameters.filter(function(p) { return p.in === 'body'; });
+    if (bodyParams.length > 1) {
+        throw new Error('Only a single body parameter allowed');
+    } else if (bodyParams.length) {
+        // Have a body parameter, special-case coercion to support formData and JSON
+        this._bodyCoercionFunc = function(req) {
+            if (req && req.body && typeof req.body === 'object') {
+                Object.keys(req.body).forEach(function(bodyKey) {
+                    req.body[bodyKey] = JSON.parse(req.body[bodyKey]);
```

Comment:
How does this work for JSON POSTs? Wouldn't this try to JSON.parse each JSON property (which is already parsed by then) as well?